test/functionalTest/cases/0876_entity_dates/filter_datecreated_sub.test

```diff
@@ -0,0 +1,307 @@
+# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Filter with dateCreated in subscriptions
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart
+
+--SHELL--
+
+#
+# 01. At t_0: Create subscription with q: dateCreated==t_2..t_4
+# 02. At t_1: Create entity E1 (not notif)
+# 03. At t_2: Create entity E2 (notif)
+# 04. At t_3: Create entity E3 (notif)
+# 05. At t_4: Create entity E4 (notif)
+# 06. At t_5: Create entity E5 (not notif)
+# 07. Get accumulator dump (3 notifications): E2, E3, E4
+#
+
+T_0=$(TZ=GMT date +"%Y-%m-%dT%H:%M:%S.00Z")
+T_2=$(TZ=GMT date +"%Y-%m-%dT%H:%M:%S.00Z" --date="2seconds")
+T_4=$(TZ=GMT date +"%Y-%m-%dT%H:%M:%S.00Z" --date="4seconds")
+
+echo "01. At t_0: Create subscription with q: dateCreated==t_2..t_4"
+echo "============================================================="
+payload='
+{
+    "subject": {
+        "entities": [
+            {
+                "idPaatern": "E.*",
+                "type": "T"
+            }
+        ],
+        "condition": {
+            "attributes": [ ],
+            "expression": {
+               "q": "dateCreated=='$T_2'..'$T_4'"
+            }
+         }
+    },
+    "notification": {
+        "callback": "http://localhost:'$LISTENER_PORT'/notify",
+        "attributes": [ ]
+    },
+    "expires": "2050-04-05T14:00:00.00Z"
+}
+'
+orionCurl --url /v2/subscriptions --payload "$payload" --json
+echo
+echo
+sleep 1s
+
```

Comment:
I would simply test it under valgrind.  (-filter filter_datecreated_sub.test)
Difficult to guess ...